jxl/src/frame/modular/tree.rs

```diff
@@ -32,6 +32,30 @@ pub enum TreeNode {
     },
 }
 
+/// Flattened tree node for optimized traversal (matches C++ FlatDecisionNode).
+/// Stores parent + info about both children to evaluate 3 nodes per iteration.
+#[derive(Debug, Clone, Copy)]
+pub(super) struct FlatTreeNode {
+    property0: i32,                    // Property to test, -1 if leaf
```

Comment:
We should leave a TODO to investigate the performance implications of using a Rust enum here.